packages/flutter_tools/lib/src/web/devfs_config.dart

```diff
@@ -0,0 +1,261 @@
+          throwToolExit('Each header entry must contain "name" and "value" keys.');
+        }
+        final dynamic name = headerMap['name'];
+        final dynamic value = headerMap['value'];
+
+        if (name is! String || value is! String) {
+          throwToolExit(
+            'Header "name" and "value" must be strings. Found name: ${name.runtimeType}, value: ${value.runtimeType}',
+          );
+        }
+        headers[name] = value;
+      }
+    }
+    if (yaml['host'] is! String && yaml['host'] != null) {
+      throwToolExit('Host must be a String. Found ${yaml['host'].runtimeType}');
+    }
+    if (yaml['port'] is! int && yaml['port'] != null) {
+      throwToolExit('Port must be an int. Found ${yaml['port'].runtimeType}');
+    }
+    if (yaml['https'] is! YamlMap && yaml['https'] != null) {
+      throwToolExit('Https must be a Map. Found ${yaml['https'].runtimeType}');
+    }
+
+    final proxyRules = <ProxyRule>[];
+    if (yaml['proxy'] != null) {
+      if (yaml['proxy'] is! YamlList) {
+        throwToolExit('Proxy must be a list. Found ${yaml['proxy'].runtimeType}');
+      }
+      final proxyList = yaml['proxy'] as YamlList;
+      for (final dynamic item in proxyList) {
+        if (item is YamlMap) {
+          final ProxyRule? rule = ProxyRule.fromYaml(item);
+          if (rule != null) {
+            proxyRules.add(rule);
+          }
+        }
+      }
+    }
+
+    return WebDevServerConfig(
+      headers: headers,
+      host: yaml['host'] as String?,
+      port: yaml['port'] as int?,
+      https: yaml['https'] == null ? null : HttpsConfig.fromYaml(yaml['https'] as YamlMap),
+      proxy: proxyRules,
+    );
+  }
+
+  final Map<String, String> headers;
+  final String? host;
+  final int? port;
+  final HttpsConfig? https;
+  final List<ProxyRule> proxy;
+
+  @override
+  String toString() {
+    return '''
+  DevConfig:
+  headers: $headers
+  host: $host
+  port: $port
+  https: $https
+  proxy: $proxy''';
+  }
+}
+
+@immutable
+class HttpsConfig {
+  const HttpsConfig({required this.certPath, required this.certKeyPath});
+
+  factory HttpsConfig.fromYaml(YamlMap yaml) {
+    if (yaml['cert-path'] is! String && yaml['cert-path'] != null) {
+      throwToolExit('Https cert-path must be a String. Found ${yaml['cert-path'].runtimeType}');
+    }
+    if (yaml['cert-key-path'] is! String && yaml['cert-key-path'] != null) {
+      throwToolExit(
+        'Https cert-key-path must be a String. Found ${yaml['cert-key-path'].runtimeType}',
+      );
+    }
+    return HttpsConfig(
+      certPath: yaml['cert-path'] as String?,
+      certKeyPath: yaml['cert-key-path'] as String?,
+    );
+  }
+
+  final String? certPath;
+  final String? certKeyPath;
+
+  @override
+  String toString() {
+    return '''
+    HttpsConfig:
+        certPath: $certPath
+        certKeyPath: $certKeyPath''';
+  }
+}
+
+T? _getOverriddenValue<T>(String filedName, T? fileValue, T? cliValue) {
```

Comment:
```suggestion
T? _getOverriddenValue<T>(String filedName, T? fileValue, {T? cliValue}) {
```

If you make at least `cliValue` named (and maybe call it "overrideWith") the invocations of this function would be easier to read (for me)

Reply:
Removed the getOrridenvalue was clunky and unnecessary